tests/testhttpserver.nim

```diff
@@ -1490,5 +1527,96 @@ suite "HTTP server testing suite":
     await server.stop()
     await server.closeWait()
 
+  asyncTest "HTTP middleware request filtering test":
+    proc init(t: typedesc[FirstMiddlewareRef],
+              data: int): HttpServerMiddlewareRef =
+      proc shandler(middleware: HttpServerMiddlewareRef, reqfence: RequestFence,
+                    handler: HttpProcessCallback2): Future[HttpResponseRef] {.
```

Comment:
```suggestion
                    nextHandler: HttpProcessCallback2): Future[HttpResponseRef] {.
```

or some other descriptive name making it clear is the responsibilty of each handler?

also, this  would ideally be described, ie the mechanism of establishing a handling priority and when handlers should call the "next" handler and when they should give a default response...